### PR TITLE
5795 Add AskUserQuestionTool to AI agent utils

### PR DESCRIPTION
**Pull request by ivicac**
Lands the server side of the Ask User Question tool. Part of #5795, a sub-issue of #4482.

The client side is already on `master` (#4799, client only). This adds the missing server half:

- `AiAgentUtilsAskUserQuestionTool`, registered as a cluster element on `AiAgentUtilsComponentHandler`
- `AiAgentSseEventType` and `AiAgentToolContextKey` in `platform-ai-api`
- `ask_user_question` routing in the SSE bridges (`WebhookTriggerController`, `AiAgentTestApiController`, `WorkflowTestApiController`)
- `resumePerform` on `AiAgentChatAction` and `AiAgentStreamChatAction`, plus the regenerated `ai-agent_v1.json` and `ai_agent-utils_v1.json` snapshots

The tool resolves its resume URL through `ActionContextAware.getResumeUrl()`.

## Behaviour at the edges

- **Non-streaming `chat` action:** questions can only be delivered over a streamed conversation. With no SSE transport in the `ToolContext`, the tool sends nothing and does not suspend. It returns guidance telling the model to proceed on stated assumptions, so the run completes instead of failing.
- **No resume URL** (e.g. editor test runs without a job): the question is still sent, but the workflow is only suspended when a resume URL exists, so no job is left waiting with nothing able to resume it.

**Known limitation, tracked separately:** until #5056 is merged, the tool sends its question and suspends, but the agent loop does not actually pause and resume on the user's answer. That behaviour is #5056's scope and is not addressed here.

## Verification

- `:server:libs:modules:components:ai:agent:utils:test` (27 tests, including 4 new `AiAgentUtilsAskUserQuestionToolTest` cases) and `:server:libs:modules:components:ai:agent:test` (27 tests) pass
- The 3 edge-case tests were confirmed failing against the pre-fix code; the happy-path test passes on both
- Checkstyle, PMD and Spotless pass for `agent-utils`

### CODE REVIEW

File: server/libs/modules/components/ai/agent/src/main/java/com/bytechef/component/ai/agent/action/AiAgentChatAction.java [hunk 1]
```diff
@@ -118,6 +130,8 @@ protected Object perform(
 
         applyStructuredOutputValidation(chatClientRequestSpec, inputParameters, context);
 
+        chatClientRequestSpec.toolContext(Map.of(AiAgentToolContextKey.ACTION_CONTEXT, context));
```
[the remark this answers is not in the document]
Comment (ivicac) [reply server/libs/modules/components/ai/agent/src/main/java/com/bytechef/component/ai/agent/action/AiAgentChatAction.java:133]: Fixed in 4b2721c7607. The non-streaming chat action has no channel to show a question on, so the tool now degrades gracefully there instead of failing: when the ToolContext carries no SSE emitter or buffered-events queue, the callback returns guidance telling the model it can't ask the user in this context and should proceed on stated assumptions. It sends no event and does not suspend, so the run completes normally. Full non-streaming support would need a new delivery contract (e.g. surfacing the questions and resume URL as the action output), which overlaps the suspend/resume rework in #5056, so it's left out of this PR. Covered by testCallWithoutEventTransportReturnsGuidanceWithoutSuspending and testCallWithoutToolContextReturnsGuidance, both confirmed failing against the previous code.

File: server/libs/modules/components/ai/agent/utils/src/main/java/com/bytechef/component/ai/agent/utils/cluster/AiAgentUtilsAskUserQuestionTool.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+                Object actionContextObj = contextMap.get(ACTION_CONTEXT);
+
+                if (!(actionContextObj instanceof ActionContext actionContext)) {
+                    throw new IllegalStateException("ActionContext not available in ToolContext");
+                }
+
+                String resumeUrl = ((ActionContextAware) actionContext).getResumeUrl();
+
+                sendQuestionEvent(toolContext, questions, resumeUrl);
```
[the remark this answers is not in the document]
Comment (ivicac) [reply server/libs/modules/components/ai/agent/utils/src/main/java/com/bytechef/component/ai/agent/utils/cluster/AiAgentUtilsAskUserQuestionTool.java:107]: Fixed in 4b2721c7607, though not with a blanket throw. getResumeUrl() is also null when there's no jobId, as in editor test runs, where the client deliberately falls back to sending the answer as a new chat message (AiAgentTestRuntimeProvider). Throwing there would break testing the tool in the editor. The real defect is suspending a job nothing can resume, so the tool now calls suspend() only when a resume URL exists. It still sends the question either way. Covered by testCallWithoutResumeUrlSendsQuestionButDoesNotSuspend, confirmed failing against the previous code (suspend was invoked), with testCallWithResumeUrlSendsQuestionAndSuspends guarding the normal path.
